### PR TITLE
benchmark sources

### PR DESCRIPTION
**Pull request by llllllllll**
Related: https://groups.google.com/forum/#!topic/zipline/pmE4-SMKg20

### CODE REVIEW

File: zipline/algorithm.py [hunk 1]
```diff
@@ -138,44 +138,74 @@ def handle_data(context, data):
...
+        algo_filename : str, optional
+            The filename for the algoscript. This will be used in exception
+            tracebacks. default: '<string>'.
+        data_frequency : {'daily', 'minute'}, optional
+            The duration of the bars.
+        capital_base : float, optional
+            How much capital to start with. default: 1.0e5
+        instant_fill : bool, optional
+            Whether to fill orders immediately or on next bar. default: False
+        asset_finder : AssetFinder, optional
```
Comment (ssanderson) [inline zipline/algorithm.py]: I don't think you can actually pass an AssetFinder right now:

https://github.com/quantopian/zipline/pull/1010/files#diff-710b8bff6f1b5d930511d643b73c6987R269

Comment (llllllllll) [inline zipline/algorithm.py]: Thanks, I didn't audit the existing entries. I will make sure there are no other invalid params there.

**Comment by llllllllll** [comment]
I have a simple case where I am passing a minutely benchmark and getting a `benchmark_period_(return|volitility)` of all `NaN`. This seems incorrect to me:

```python
import numpy as np
import pandas as pd

from zipline.algorithm import TradingAlgorithm
from zipline.api import get_datetime
from zipline.sources import BenchmarkSource
from zipline.finance.trading import TradingEnvironment

dates = TradingEnvironment().minutes_for_days_in_range(
    '2014-01-06',
    '2014-01-10',
)
data = pd.Panel({
    0: pd.DataFrame(
        {'price': np.arange(len(dates)), 'volume': np.arange(len(dates))},
        index=dates,
    ),
})
bench = BenchmarkSource(pd.Series(np.arange(len(dates)), index=dates))


def initialize(context):
    pass


def handle_data(context, data):
    print(get_datetime())


results = TradingAlgorithm(
    start=dates[0],
    end=dates[-1],
    data_frequency='minute',
    emission_rate='minute',
    initialize=initialize,
    handle_data=handle_data,
).run(data, benchmark_return_source=bench)
```

**Comment by llllllllll** [comment]
we have changed the source api

**Pull request closed without merging**